src/lib/menu/menu-directive.ts

```diff
@@ -81,6 +100,12 @@ export class MdMenu implements AfterContentInit, MdMenuPanel, OnDestroy {
     }
   }
 
+  ngOnChanges(changes: SimpleChanges) {
+    if (changes['xPosition'] || changes['yPosition']) {
+      this.setPositionClasses();
```

Comment:
Not sure I understood the question. The setters aren't dealing with the classes at all.